src/transformers/modeling_dpr.py

```diff
@@ -0,0 +1,551 @@
+# coding=utf-8
+# Copyright 2018 DPR Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+""" PyTorch DPR model for Open Domain Question Answering."""
+
+
+import collections
+import logging
+from typing import List, Optional, Tuple
+
+import torch
+from torch import Tensor as T
+from torch import nn
+from torch.serialization import default_restore_location
+
+from .configuration_dpr import DprConfig
+from .file_utils import add_start_docstrings
+from .modeling_bert import BertConfig, BertModel
+from .modeling_utils import PreTrainedModel
+
+
+logger = logging.getLogger(__name__)
+
+DPR_PRETRAINED_MODEL_ARCHIVE_LIST = [
+    "dpr-base-uncased",
+]
+
+
+#############
+# BertEncoder
+#############
+
+
+class DprBertEncoder(BertModel):
```

Comment:
Don't think we should use abstraction here. I think it's much better if you save `BertModel` as a layer:

```python 
self.bert_encoder = BertModel(...)
```

We always use composition in cases where a model is extended by a layer or two IMO. What do you think @sshleifer @LysandreJik ?

Reply:
I prefer inheritance here because then the caller code doesn't have to write `self.bert_encoder.blah` all over the place.

I'd also just call it `DPREncoder`

Reply:
I don't have a strong opinion between the two as the case isn't complicated (either way the code is readable).
However I'd still like to keep inheritance here for one reason: it is the format used in the official code and therefore it allows to easily load the weights from the official repo.

Reply:
I agree with Patrick here that we usually have `BertModel` as an attribute. It doesn't seem worth it to me to do it differently here.

Reply:
I managed to do the conversion !
I'll change it to composition then.

Reply:
I just did the change, could you check if that;s fine now ?